src/carousel.component.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { FaChevronRight, FaChevronLeft } from 'react-icons/fa';
+
+const SlideImage = styled.img`
+  width: 200px;
+  height: 100px;
+  object-fit: cover;
+`;
+
+const StyledSlider = styled.div`
+  display: flex;
+  position: relative;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  .slides {
+    margin: 0 auto;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+  }
+`;
+
+function CarouselComponent({slides}) {
+  const [current, setCurrent] = useState(0);
+  const length = slides.length;
+
+  const nextSlide = () => {
+    setCurrent(current === length - 1 ? 0 : current + 1);
+  };
+
+  const prevSlide = () => {
+    setCurrent(current === 0 ? length - 1 : current - 1);
+  };
```

Comment:
Aquí sugeriría usar el setState con un callback, para evitar bugs donde tu state internamente ya se actualizó, pero tu componente aún no se ha rendereado con el nuevo valor. Es un edge case que igual no te encuentras, pero cuando te lo topas es difícil de debuggear, así que si puedes evitarlo, evítalo.

```suggestion
  const nextSlide = () => {
    setCurrent((current) => current === length - 1 ? 0 : current + 1);
  };

  const prevSlide = () => {
    setCurrent((current) => current === 0 ? length - 1 : current - 1);
  };
```